solver/llbsolver/solver.go

```diff
@@ -175,7 +175,7 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 			}
 			inp.Ref = workerRef.ImmutableRef
 
-			dtbi, err := buildinfo.Merge(ctx, res.BuildInfo(), inp.Metadata[exptypes.ExporterImageConfigKey])
+			dtbi, err := buildinfo.Get(ctx, req, res.BuildSources(), inp.Metadata[exptypes.ExporterImageConfigKey])
```

Comment:
I don't think this works with the `client.Build` command where the requests are sent with gateway API. I think frontend could probably still initially set and if it does then we can clear it here. Or do we want to handle a frontend even it has not been updated? In that case, I think the attrs should be wrapped in the `llbbridge.Result` handler.

Reply:
This does not seem to be addressed. I don't have a strong preference if the attr should be added inside frontend or in LLB (although would be good if frontend at least has some control of it, maybe discussion for later). But it should definitely work with gateway API as well.

Reply:
Yes I'm still exploring the gateway side to see how I can handle that effectively both ways.

Reply:
I think if you wrap the bridge result then the non-gateway part will also just automatically start to work.